### PR TITLE
Add flag to agent to control writing job history to ZK

### PR DESCRIPTION
**Pull request by davidxia**
Default is true which is previous behavior.

### CODE REVIEW

**Comment by mattnworb** [comment]
👍

**Comment by davidxia** [comment]
@mattnworb @rohansingh I also added javadoc to `OldJobReaper` since it will be affected if this flag is enabled.

**Comment by davidxia** [comment]
cc @saunakchakrabarti

**Comment by davidxia** [comment]
Was thinking about adding a test for the flag. The relevant class is `JobHistoryTest` which is an IT. But there's not really a good way to test that no event will ever be sent. I didn't feel like sleeping a bunch and then checking that there's no events would be a worthwhile test.

**Comment by mattnworb** [comment]
@davidxia I think it would be sufficient to add a unit test of `ZooKeeperAgentModel` that sees what it does when the TaskHistoryWriter instance is null

**Comment by davidxia** [comment]
@mattnworb I renamed the flag and added a test in b9bb5f3.

File: helios-services/src/main/java/com/spotify/helios/agent/AgentConfig.java [hunk 1]
```diff
@@ -359,6 +360,15 @@ public AgentConfig setFfwdConfig(FastForwardConfig config) {
     return this;
   }
 
+  public boolean getNoJobHistory() {
```
Comment (mattnworb) [inline helios-services/src/main/java/com/spotify/helios/agent/AgentConfig.java]: `isJobHistoryDisabled()` would be a more readable way to phrase this

File: helios-services/src/main/java/com/spotify/helios/agent/AgentParser.java [hunk 2]
```diff
@@ -265,6 +267,11 @@ protected void addArgs(final ArgumentParser parser) {
         .help("ZooKeeper agent password (for ZooKeeper ACLs). If the "
               + ZK_AGENT_PASSWORD_ENVVAR
               + " environment variable is present this argument is ignored.");
+
+    noJobHistory = parser.addArgument("--no-job-history")
```
Comment (mattnworb) [inline helios-services/src/main/java/com/spotify/helios/agent/AgentParser.java]: `--disable-job-history` might be more intuitive ?

**Comment by mattnworb** [comment]
👍  regardless of how you want to handle the above comments

File: helios-services/src/main/java/com/spotify/helios/agent/AgentParser.java (hunk 2 above)
Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/agent/AgentParser.java]: Yea, I was trying to save chars.
